native/packages/react-native-bpk-component-pagination-dots/src/BpkPaginationDots.js

```diff
@@ -0,0 +1,206 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import { StyleSheet, View, ViewPropTypes } from 'react-native';
+import PropTypes from 'prop-types';
+import { setOpacity } from 'bpk-tokens';
+import {
+  borderRadiusSm,
+  colorGray900,
+  spacingSm,
+  spacingMd,
+  spacingLg,
+} from 'bpk-tokens/tokens/base.react.native';
+import BpkPaginationDotsIndicator, {
+  INDICATOR_SIZES,
+} from './BpkPaginationDotsIndicator';
+
+const styles = StyleSheet.create({
+  wrapper: {
+    alignItems: 'center',
+    backgroundColor: setOpacity(colorGray900, 0.3),
+    borderRadius: borderRadiusSm,
+    flexDirection: 'row',
+    height: spacingLg,
+    justifyContent: 'center',
+    paddingHorizontal: spacingSm + spacingMd,
+    paddingVertical: spacingMd,
+    width: spacingLg * 3,
+  },
+});
+
+const indicatorMappings = {
+  // When selectedPageIndex is 0, 1 or 2.
+  // Looks like 'OOOo.'
+  start: [
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.md,
+    INDICATOR_SIZES.sm,
+    INDICATOR_SIZES.invisible,
+  ],
+  // When selectedPageIndex is 3.
+  // Looks like 'oOOOo.'
+  closeToStart: [
+    INDICATOR_SIZES.md,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.md,
+    INDICATOR_SIZES.sm,
+  ],
+  // When selectedPageIndex is greater than 3 but less than pageCount - 2.
+  // Looks like '.oOOo.'
+  middle: [
+    INDICATOR_SIZES.sm,
+    INDICATOR_SIZES.md,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.md,
+    INDICATOR_SIZES.sm,
+  ],
+  // When selectedPageIndex is pageCount - 2.
+  // Looks like '.oOOo'
+  closeToEnd: [
+    INDICATOR_SIZES.invisible,
+    INDICATOR_SIZES.sm,
+    INDICATOR_SIZES.md,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.md,
+  ],
+  // When selectedPageIndex is pageCount - 1.
+  // Looks like '.oOOO'
+  end: [
+    INDICATOR_SIZES.invisible,
+    INDICATOR_SIZES.sm,
+    INDICATOR_SIZES.md,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+    INDICATOR_SIZES.base,
+  ],
+};
+
+const getIndicators = (pageCount, selectedPageIndex) => {
+  const createIndicatorsFromSizes = (sizesArray, selectedIndex) =>
+    sizesArray.map((indicatorSize, index) => (
+      <BpkPaginationDotsIndicator
+        key={index.toString()}
```

Comment:
Might not want to do this as you rely on element reuse for the animations and this might break that. Frankly I don't know enough about React internals to say

Reply:
I verfied it and doign this does break animations

Reply:
Breaks it in what way? As far as I can see it works fine..

Reply:
Oh wait, do you mean adding `-size` breaks it?

Reply:
Yeah adding the size does break the animation because the underlying nodes don't get reused